enhancements/console/customize-project-roles.md

```diff
@@ -0,0 +1,162 @@
+---
+title: customize-project-roles
+authors:
+  - "@jerolimov"
+reviewers:
+  - "@spadgett"
+  - "@cvogt"
+approvers:
+  - "@spadgett"
+  - "@cvogt"
+creation-date: 2021-02-25
+last-updated: 2021-02-25
+status: implementable
+---
+
+# Customize Project Access Roles
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The OpenShift Developer Console provides a simple way to share access to the selected project. Users can be assigned to specific roles under the Project menu item on the Project access tab. Currently only the roles "Admin", "Edit" and "View" are selectable.
+
+OpenShift Console 3.11 allowed customers to customize the list of roles visible in the membership page. This enhancement will provide equivalent functionality in 4.x.
+
+Cluster admins want give project admins only access to a customized list of predefined roles to force compliance rules or provide more options.
+
+## Motivation
+
+Customers with customized roles want to provide easy access for their project admins to restrict access within a namespace.
+
+### Goals
+
+- Add configuration options to allow cluster admins to customize the project access roles.
+- Provide a list of default values to the admin.
+
+### Non-Goals
+
+- Introducing a UI to configure this project access roles without a YAML editor.
+
+## Proposal
+
+### Extending the Console CRD
+
+Similar to [Customize Developer Catalog Categories](./catalog-categories.md) we extend the existing `operator.openshift.io/v1` / `Console` CRD. It provides already a `spec.customization` area where we can add a list of roles.
+
+The list of roles should link to `rbac.authorization.k8s.io/v1` / `Role` or `ClusterRole`.
```

Comment:
I would limit this to `ClusterRole` for now. I'm not sure it's a good idea to allow this to be configured for `Role` since `Roles` with the same names in different namespaces can have totally different permission. The cluster admin can't know what these will be when configuring the option. Also there is no way in the current UI to differentiate between `Role` and `ClusterRole`, so as a project admin you won't know exactly what permission you're granting.

Reply:
We can show R or CR in the dropdown. But I think ClusterRole is fine because the admin could not predict what project admins allow with their namespaced `Role` -- as you mentioned. So, this is our answer for just a list of strings, right? No label, no kind is required here. Just `projectAccessRoles` or `clusterRoleOptions` or `projectAccess.availableClusterRoles` ...

Reply:
Also noticed that the dropdown has `Role` label and also the help-text above is linking to `Role` docs. If we want to just support `ClusterRole` for no we should make that clear, for both mentioned elements.

Reply:
Good point, then we should also link directly to ClusterRoles and ClusterRoleBindings in the help text above the input. Sam, Christian, @serenamarie125 fine with that?